scripts/install

```diff
@@ -38,76 +42,76 @@ grep -q -R 'jessie-backports' /etc/apt/sources.list{,.d} || {
 
 # Initialize database
 ynh_mysql_create_db "$dbname" "$dbuser" "$dbpass"
 ynh_mysql_connect_as "$dbuser" "$dbpass" "$dbname" \
-    < "${TMPDIR}/SQL/mysql.initial.sql"
+    < "${final_path}/SQL/mysql.initial.sql"
 
 # Copy and set Roundcube configuration
-rc_conf="${TMPDIR}/config/config.inc.php"
+rc_conf="${final_path}/config/config.inc.php"
 cp ../conf/config.inc.php "$rc_conf"
-sed -i "s/#DESKEY#/${deskey}/g" "$rc_conf"
-sed -i "s/#DBUSER#/${dbuser}/g" "$rc_conf"
-sed -i "s/#DBPASS#/${dbpass}/g" "$rc_conf"
-sed -i "s/#DBNAME#/${dbname}/g" "$rc_conf"
+ynh_substitute_char "#DESKEY#" "$deskey" "$rc_conf"
+ynh_substitute_char "#DBUSER#" "$dbuser" "$rc_conf"
+ynh_substitute_char "#DBPASS#" "$dbpass" "$rc_conf"
+ynh_substitute_char "#DBNAME#" "$dbname" "$rc_conf"
 
 # Install files and set permissions
-sudo mv "$TMPDIR" "$DESTDIR"
-sudo mkdir -p "${DESTDIR}/logs" "${DESTDIR}/temp"
-sudo chown -R www-data: "$DESTDIR"
-
-# Install some plugins manually
-sudo cp -r ../sources/plugins/ldapAliasSync "${DESTDIR}/plugins"
-sudo chown -R www-data: "${DESTDIR}/plugins/ldapAliasSync"
-installed_plugins=" 'ldapAliasSync',"
+sudo mkdir -p "${final_path}/logs" "${final_path}/temp"
 
 # Install additional plugins
-exec_composer www-data "$DESTDIR" require \
-    "johndoh/contextmenu dev-release-2.1" \
-    "sblaisot/automatic_addressbook"
+exec_composer admin "$final_path" require \
+    "johndoh/contextmenu dev-master" \
+    "sblaisot/automatic_addressbook dev-master"
 installed_plugins+=" 'contextmenu', 'automatic_addressbook',"
 
-# Instal CardDAV plugin
+# Install CardDAV plugin
 if [[ $with_carddav -eq 1 ]]; then
-  install_carddav "$DESTDIR" \
+  install_carddav "$final_path" admin \
     && installed_plugins+=" 'carddav'," \
     || echo "Unable to install CardDAV plugin" >&2
 fi
 
+# Install Enigma plugin
+if [[ $with_enigma -eq 1 ]]; then
+  sudo cp -a "$final_path/plugins/enigma/config.inc.php.dist" "$final_path/plugins/enigma/config.inc.php" \
+    && installed_plugins+=" 'enigma'," \
+    || echo "Unable to install Enigma plugin" >&2
+fi
+
 # Update Roundcube configuration
 sudo sed -i "s#^\s*// installed plugins#&\n ${installed_plugins}#" \
-    "${DESTDIR}/config/config.inc.php"
-
-# Copy and set nginx configuration
-nginx_conf="/etc/nginx/conf.d/${domain}.d/${app}.conf"
-sed -i "s@#PATH#@${path:-/}@g"     ../conf/nginx.conf
-sed -i "s@#DESTDIR#@${DESTDIR}/@g" ../conf/nginx.conf
-sudo cp ../conf/nginx.conf "$nginx_conf"
-
-# Copy and set php-fpm configuration
-phpfpm_conf="/etc/php5/fpm/pool.d/${app}.conf"
-sed -i "s@#POOLNAME#@${app}@g"     ../conf/php-fpm.conf
-sed -i "s@#DESTDIR#@${DESTDIR}/@g" ../conf/php-fpm.conf
-sudo cp ../conf/php-fpm.conf "$phpfpm_conf"
-sudo chown root: $phpfpm_conf
-sudo chmod 644 $phpfpm_conf
+    "${final_path}/config/config.inc.php"
+
+# Update javascript dependencies
+pushd ${final_path}
+sudo /usr/bin/php -q ./bin/install-jsdeps.sh
+popd
+
+# Change owner final_path directory
+sudo chown -R $app: "${final_path}"
```

Comment:
Maybe here some directories could stay with root owner if the app doesn't need to modify/creates files...?